### PR TITLE
Catch crashes and write minidumps on windows

### PR DESCRIPTION
**Pull request by dimhotepus**
Kinda hard to debug issues without crash dumps on hands.  Add crash dumps writing support on Windows. Makes resolving issues like https://github.com/dreamstalker/rehlds/issues/523, https://github.com/dreamstalker/rehlds/issues/698, https://github.com/dreamstalker/rehlds/issues/699 easier.

Crash dump is written to file in the same directory as crashed executable.  Crash dump file name format: `<exe name>_<app version>_crash_<date YYYYMMDD>_<time HHMMSS>_<unique counter>.mdmp`
for example, `hlds_3.6.0.675-dev+m_crash_20190702_005019_1.mdmp`

Command line option `-nominidumps` disables mini dump writing facility (same behavior as in Source games). Also, mini dumps are not written when debugger is attached.

Uses `DbgHelp.dll` to write minidumps, should be present on all Windows versions, but **REQUIRES checking** on Windows XP+. Tested on Windows 10 18362.

Tries to write minidump with data segments, handles, threads and indirectly referenced memory first.
As a fallback, writes minidump with data segments only.

### CODE REVIEW

**Comment by dimhotepus** [comment]
@WPMGPRoSToTeMa, @theAsmodai Can you, please, suggest how to fix missing symbols during link step?

Locally all builds and works ok (but i've converted solution to vs 2019 format + c++14).

File: rehlds/common/minidump.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			local_time_now->tm_mday,        /* day of month (1 - 31) */
+			local_time_now->tm_hour,        /* hour (0 - 23) */
+			local_time_now->tm_min,         /* minutes (0 - 59) */
+			local_time_now->tm_sec,         /* seconds (0 - 59) */
+			++minidumps_written_count);     // ensures the filename is unique
+
+		minidump_file = CreateFileW(file_name, GENERIC_WRITE, FILE_SHARE_WRITE,
+									nullptr, CREATE_ALWAYS,
+									FILE_ATTRIBUTE_NORMAL, nullptr);
+		error_code = !!minidump_file ? S_OK : GetLastHresult();
```
Comment (tsarevichdm) [inline rehlds/common/minidump.cpp]: Please, fix check, `minidump_file` will be `INVALID_HANDLE_VALUE` when `CreateFileW` failed.

See https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew#return-value

Comment (dimhotepus) [reply rehlds/common/minidump.cpp]: Done, thank you.

**Comment by In-line** [comment]
Seems to be okay at the first glance, build should be fixed and I think this can be merged hopefully. I don't run windows, to manually test.

**Comment by dimhotepus** [comment]
@In-line Done, thanks!

**Comment by dimhotepus** [comment]
@s1lentq Thank you for fixing problems. Commit hash is good idea, too.

**Pull request merged**